test/e2e/specs/specs-playwright/wp-domains__add.ts

```diff
@@ -76,7 +76,15 @@ describe( DataHelper.createSuiteTitle( 'Domains: Add to current site' ), functio
 	} );
 
 	it( 'Make purchase', async function () {
-		await cartCheckoutPage.purchase();
+		await Promise.all( [
+			page.waitForNavigation( {
+				url: '**/checkout/thank-you/**',
+				waitUntil: 'networkidle',
+				// Sometimes the testing domain third party system is really slow. It's better to wait a while than to throw a false positive.
+				timeout: 90 * 1000,
+			} ),
+			cartCheckoutPage.purchase(),
```

Comment:
Do you think it would be better to pass the `waitForNavigation` options to the `purchase` method since [it's also waiting for the nav event](https://github.com/Automattic/wp-calypso/blob/9e1f09a8c8ab1a84f9df140fb581d99527732797/packages/calypso-e2e/src/lib/pages/cart-checkout-page.ts#L241)?

Reply:
Ahh, this is such a good question! @worldomonation and I have actually run into this before and had this very discussion! This happens more than I initially would think, where a specific POM or test needs to do a more specific `waitForNavigation` than other places.

At the time, our initial conclusion was to let those one-off places just wrap the call with their more specific `waitForNavigation`. It doesn't seem to trip up Playwright at all - if there are two unresolved navigation promises, and we navigate, it seems to satisfy both, which is nice!

Letting the caller add the extra promise wrap simplifies the function signature a lot - we were finding ourselves having to add a lot of optional `options` style parameters and it didn't seem quite as clean.

Thoughts on that? Anything to add @worldomonation

Reply:
> Do you think it would be better to pass the waitForNavigation options to the purchase method since it's also waiting for the nav event?

In general, yes. The library files should work for the broadest possible situations; however in certain cases (such as this), when we are looking to narrow the behavior down to something more specific, we've chosen to expect the test writer to account for the narrowing of behavior.

In this case specifically, the navigation timeout is to be extended **only** for these two steps and nowhere else.

As @dpasque mentioned, Playwright typically does not run into issues with multiple stacked `waitFor` calls, though to avoid hard-to-debug race issues it's generally preferable not to stack them. It works in this case because the `waitForNavigation` at the spec level take longer to resolve than the one contained in the library file.